subsystems/TypeHandlerLibrary/src/main/java/org/terasology/persistence/typeHandling/coreTypes/factories/CollectionTypeHandlerFactory.java

```diff
@@ -24,7 +24,8 @@
 public class CollectionTypeHandlerFactory implements TypeHandlerFactory {
     private static final Logger LOGGER = LoggerFactory.getLogger(CollectionTypeHandlerFactory.class);
 
-    private ConstructorLibrary constructorLibrary;
+    // PMD and intellij cannot judge if this is good or not. suppress.
+    @SuppressWarnings("PMD.UnusedLocalVariable") private ConstructorLibrary constructorLibrary;
```

Comment:
I'm still questioning whether we want to suppress this. Again, the age of the code is not a valid reason IMO.

Reply:
I believe, we can remove this field from the `CollectionTypeHandlerFactory` class.
I tested it locally and didn't see any issues. My assumption is that this is a left-over from refactoring `ConstructorLibrary#getCollectionCopyConstructor` to a static method. This allowed the call in `CollectionTypeHandlerFactory#create` to use the method without instantiating `ConstructorLibrary`.